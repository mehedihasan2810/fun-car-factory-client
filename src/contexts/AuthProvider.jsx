import { createContext, useEffect, useState } from "react";
import {
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  signInWithEmailAndPassword,
  signOut,
  GoogleAuthProvider,
  updateProfile,
  signInWithPopup,
} from "firebase/auth";
import { auth } from "../configs/firebase";

export const AuthContext = createContext();

const gooleProvider = new GoogleAuthProvider();

const AuthProvider = ({ children }) => {
  const [currentUser, setCurrentUser] = useState(null);
  const [isAuthLoading, setIsAuthLoading] = useState(true);

  const signUp = (email, password) => {
    return createUserWithEmailAndPassword(auth, email, password);
  };

  const updateUserProfile = (user, name, url) => {
    return updateProfile(user, {
      displayName: name,
      photoURL: url,
    });
  };

  const signIn = (email, password) => {
    return signInWithEmailAndPassword(auth, email, password);
  };

  const googleSignIn = () => {
    return signInWithPopup(auth, gooleProvider);
  };

  const logOut = () => {
    return signOut(auth);
  };

  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (user) => {
      setCurrentUser(user);
      setIsAuthLoading(false);
    });

    return () => {
      unsubscribe();
    };
  }, []);

  return (
    <AuthContext.Provider
      value={{
        googleSignIn,
        signUp,
        updateUserProfile,
        signIn,
        logOut,
        currentUser,
        isAuthLoading,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

export default AuthProvider;
