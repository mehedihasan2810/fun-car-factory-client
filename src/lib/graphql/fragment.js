import gql from "graphql-tag";

export const CORE_CARS_FIELD = gql`
  fragment CoreCarsField on Car {
    id
    name
    price
    url
    rating
    category
    quantity
  }
`;
