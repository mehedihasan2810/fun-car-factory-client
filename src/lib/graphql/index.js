import {
  ApolloClient,
  InMemoryCache,
  createHttpLink,
  from,
} from "@apollo/client";
import { setContext } from "@apollo/client/link/context";
import { onError } from "@apollo/client/link/error";
import { signOut } from "firebase/auth";
import Cookies from "js-cookie";
import { auth } from "../../configs/firebase";

// Create an in-memory cache
export const cache = new InMemoryCache();

// Create an HTTP link to the GraphQL server
const httpLink = createHttpLink({
  uri: "https://fun-car-factory-server.vercel.app/graphql",
  // uri: "http://localhost:4000/graphql",
});

// Set the authorization context based on the user's token
const authLink = setContext((_, { headers }) => {
  const token = Cookies.get("token");

  return {
    headers: {
      ...headers,
      authorization: token ? `Bearer ${token}` : "",
    },
  };
});

// Handle errors, including authentication errors
const errorLink = onError(async ({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    if (graphQLErrors[0]?.message === "User is not authenticated") {
      // Sign out the user, remove the token, and clear the Apollo Client store
      await signOut(auth);

      Cookies.remove("token");

      apolloClient.clearStore();
    }

    // Log GraphQL errors
    graphQLErrors.forEach(({ message }) =>
      console.error(`[GraphQL error]: Message: ${message}`)
    );
  }
  if (networkError) console.error(`[Network error]: ${networkError}`);
});

// Create the Apollo Client with the configured link and cache
export const apolloClient = new ApolloClient({
  // link: authLink.concat(errorLink, httpLink),
  link: from([authLink, errorLink, httpLink]),
  cache,
});
