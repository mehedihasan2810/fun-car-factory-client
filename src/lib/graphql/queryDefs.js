import { gql } from "@apollo/client";

export const GET_CARS = gql`
  query GetCars {
    getCars {
      id
      name
      price
      url
      rating
    }
  }
`;

export const GET_CAR = gql`
  query GetCar($id: String!) {
    getCar(id: $id) {
      id
      name
      price
      rating
      url
      sellerName
      email
      category
      description
    }
  }
`;
