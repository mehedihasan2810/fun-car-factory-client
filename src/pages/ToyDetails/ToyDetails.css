.details-container {
  display: grid;
  grid-template-columns: 2fr 1fr;
  width: min(100vw, 1200px);
  color: var(--primary-sky-50);
  gap: 3rem;
  margin-block: 200px;
  margin-inline: auto;
  color: var(--base-gray-300);
  padding-inline: 8px;
}

@media (max-width: 768px) {
  .details-container {
    grid-template-columns: 1fr;
    margin-block: 120px;
  }
}

.details-container .img-galary {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 0.5rem;
}

.details-container img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.details-container .details {
  width: max-content;
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.details-container h4 {
  font-size: 1.4rem;
  font-weight: 600;
  color: var(--base-gray-50);
}

.details-container .seller {
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.details-container .rating {
  display: flex;
  align-items: center;
  gap: 5px;
  font-size: 15px;
  color: var(--base-gray-50);
}

.toy-details-btns {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.toy-details-btns > button {
  padding: 1.2rem 1.5rem;
  background-color: var(--primary-sky-600);
  color: var(--primary-sky-50);
  border: 0;
  border-radius: 100vw;
  cursor: pointer;
}

.toy-details-btns > button:hover {
  background-color: var(--primary-sky-500);
}
