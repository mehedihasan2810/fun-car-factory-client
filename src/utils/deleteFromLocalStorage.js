const deleteFromLocalStorage = (key, id) => {
  if (localStorage.getItem(key)) {
    const items = JSON.parse(localStorage.getItem(key));
    const filteredItems = items.filter((item) => id !== item);
    localStorage.setItem(key, JSON.stringify(filteredItems));
  }
};

export default deleteFromLocalStorage;
