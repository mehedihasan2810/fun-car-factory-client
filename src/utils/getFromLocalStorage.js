const getFromLocalStorage = (key) => {
  return localStorage.getItem(key);
};

export default getFromLocalStorage;
