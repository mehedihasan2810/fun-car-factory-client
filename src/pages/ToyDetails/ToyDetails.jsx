import { useLoaderData } from "react-router-dom";
import { FaStar } from "react-icons/fa";
import "./ToyDetails.css";
const ToyDetails = () => {
  const toy = useLoaderData();
  console.log(toy);
  return (
    <div className="center-container">
      <div className="details-container">
        <div className="img-galary">
          <img src={toy.url} alt="" />
          <img src={toy.url} alt="" />
          <img src={toy.url} alt="" />
          <img src={toy.url} alt="" />
        </div>
        <div className="details">
          <h4>{toy.name}</h4>
          <p className="price">{toy.price}$</p>
          <p>Category: {toy.category}</p>
          <p>{toy.quantity} pieces available</p>

          <div className="seller">
            <p>Seller: {toy.sellerName}</p>
            <p>Email: {toy.email}</p>
          </div>
          <p className="rating">
            <FaStar style={{ color: "rgb(233, 142, 22)" }} /> {toy.rating}
            /5(100)
          </p>
          <p className="desc">{toy.description}</p>
        </div>
      </div>
    </div>
  );
};

export default ToyDetails;
