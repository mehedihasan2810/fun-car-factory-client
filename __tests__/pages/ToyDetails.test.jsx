import { ApolloProvider } from "@apollo/client";
import { describe, expect, screen, test } from "../utils/utils";
import { apolloClient } from "../../src/lib/graphql";
import ToyDetails from "../../src/pages/ToyDetails/ToyDetails";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import AuthProvider from "../../src/contexts/AuthProvider";
import { carsMockData } from "../mocks/carsMockData";

describe("<ToyDetails />", () => {
  test("should render, fetch single data and match snapshot succesfully", async () => {
    const id = "64671bf3e81294b16783e451";
    const { asFragment } = render(
      <MemoryRouter initialEntries={[`/toy-details/${id}`]}>
        <AuthProvider>
          <ApolloProvider client={apolloClient}>
            <Routes>
              <Route path="/toy-details/:id" element={<ToyDetails />} />
            </Routes>
          </ApolloProvider>
        </AuthProvider>
      </MemoryRouter>
    );

    expect(
      screen.getByText("toy-details-loading-skeleton")
    ).toBeInTheDocument();

    await waitForElementToBeRemoved(() =>
      screen.getByText("toy-details-loading-skeleton")
    );

    const filteredData = carsMockData.find((car) => car.id === id);

    expect(
      screen.getByRole("heading", { name: filteredData.name, level: 4 })
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
