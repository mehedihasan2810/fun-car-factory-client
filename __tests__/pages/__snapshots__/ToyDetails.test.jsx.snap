// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ToyDetails /> > should render  succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ToyDetails /> > should render succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ToyDetails /> > should render,  succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ToyDetails /> > should render, fetch  succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ToyDetails /> > should render, fetch single data and  succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ToyDetails /> > should render, fetch single data and match snapshot succesfully 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <div
      class="img-galary"
    >
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
      <img
        alt=""
        src="https://images.pexels.com/photos/163836/vw-bulli-meadow-peace-163836.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
      />
    </div>
    <div
      class="details"
    >
      <h4>
        TaTa super bus
      </h4>
      <p
        class="price"
      >
        12$
      </p>
      <p>
        Category: Bus
      </p>
      <p>
         pieces available
      </p>
      <div
        class="seller"
      >
        <p>
          Seller: mehedi
        </p>
        <p>
          Email: mehedimiah47@gmail.com
        </p>
      </div>
      <p
        class="rating"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(233, 142, 22);"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
         5/5(100)
      </p>
      <p
        class="desc"
      >
        this is tata super bus
      </p>
      <div
        class="toy-details-btns"
      >
        <button>
          Buy Now
        </button>
        <button>
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
